lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -786,14 +786,17 @@ def remove_vm(self, vm):
             return {'changed': True, 'failed': False}
 
     def configure_guestid(self, vm_obj, vm_creation=False):
+        # guest_id is not required when using templates
+        if self.should_deploy_from_template():
```

Comment:
sorry, but if i have a template debian 8 and debian 9 is released, maybe i prefer to create debian 8 from debian 9 and change guestid. It's not the good solution. I think it's better to check the guestId value